File: API/API/Models/Booking.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace API.Models
{
    public class Booking
    {
        public int Id { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public StatusBooking Status { get; set; }
        public double TotalPrice { get; set; }
        public int UserId { get; set; }
        [JsonIgnore]
        public User User { get; set; }
        public List<ListRoom> ListRooms { get; set; }

        //public int QuantityRoom { get; set; }
        //public int QuantityRoomExcess { get; set; }
        //public int RoomId { get; set; }
        //[JsonIgnore]
        //public Room Room { get; set; }
        public DateTime DateCreated { get; set; }
        public StatusCheckIn StatusCheckIn { get; set; }
        public DateTime CheckInTime { get; set; }
    }
    public class ListRoom
    {
        public int Id { get; set; }
        public int RoomId { get; set; }
        [JsonIgnore]
        public Room Room { get; set; }
        public int BookingId { get; set; }
        [JsonIgnore]
        public Booking Booking { get; set; }
        public int QuantityRoom { get; set; }
        public int QuantityRoomExcess { get; set; }
    }
}

File: Fontend/src/components/models/booking.ts
export interface Booking {
  id: number;
  listRooms:  [
    {
      id: number;
      roomId: number;
      bookingId: number;
      quantityRoom: number;
      quantityRoomExcess: number;
    }
  ];
  start: string;
  end: string;
  status: number;
  totalPrice: number;
  userId: number;
  dateCreated: string;
  statusCheckIn: number;
  checkInTime: string;
}

export interface BookingPayment {
  id: number;
  paymentDate: Date;
  status: number;
  bookingId: number;
  paymentIntentId0?: string | null;
  clientSecret0?: string | null;
  paymentIntentId1?: string | null;
  clientSecret1?: string | null;
  paymentIntentId2?: string | null;
  clientSecret2?: string | null;
}

export interface BookingPackage {
  id: number;
  listPackages:  [
    {
      id: number;
      packageId: number;
      bookingPackageId: number;
      quantity: number;
      start: string;
      end: string;
      checkInTime: string;
      checkInDate: number;
    }
  ];
  totalPriceBookingPackage: number;
  dateCreated: string;
  status: number;
  userId: number;
}

export interface BookingPackagePayment {
  id: number;
  paymentDate: Date;
  status: number;
  bookingPackageId: number;
  paymentIntentId0?: string | null;
  clientSecret0?: string | null;
  paymentIntentId1?: string | null;
  clientSecret1?: string | null;
  paymentIntentId2?: string | null;
  clientSecret2?: string | null;
}